homeassistant/components/zamg/config_flow.py

```diff
@@ -111,26 +104,17 @@ async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
             longitude = config.get(CONF_LONGITUDE) or self.hass.config.longitude
             station_id = await self._client.closest_station(latitude, longitude)
 
```

Comment:
```suggestion
            if station_id is None:
                LOGGER.error("Error finding closest station")
                errors: dict[str, Any] = {}
                errors["base"] = "cannot_connect"
                return self.async_abort(
                    reason="cannot_connect", description_placeholders=errors
                )

```
Did you mean a catch like this for a failed api call?

Reply:
No, I'm talking about the call to `self._client.zamg_stations` on line 98. We can't test if something is `in None`.

Reply:
And it looks like it can raise `asyncio.TimeoutError`.

https://github.com/killer0071234/python-zamg/blob/05f680a9b43de7cefb92bbe49388605296833205/src/zamg/zamg.py#L59

Reply:
The `session.get` call can raise aiohttp exceptions too.

Reply:
> And it looks like it can raise asyncio.TimeoutError.

That's true, i will update `python-zamg` to catch asyncio errors

Reply:
I suggest raising a library specific exception instead of returning an invented default value. Let the caller decide what to do if the call fails. It's more clear to raise a custom exception than to return an unexpected return value.

https://github.com/killer0071234/python-zamg/blob/8b9bb470795497957df3dd448350e44cb6d3fbc4/src/zamg/zamg.py#L92

Why should the library check for None values for latitude and longitude when they are typed as float?

https://github.com/killer0071234/python-zamg/blob/8b9bb470795497957df3dd448350e44cb6d3fbc4/src/zamg/zamg.py#L96

This can't happen:
https://github.com/killer0071234/python-zamg/blob/8b9bb470795497957df3dd448350e44cb6d3fbc4/src/zamg/zamg.py#L99

Reply:
Yes you are right to have Exceptions instead of some strange values is better.
Thanks for the hint, i will dig into it.